### PR TITLE
Adds a lava canyon as a new lava planet ruin

### PR DESCRIPTION
**Pull request by MemeSnorfer**
## About The Pull Request

Adds a lava canyon ruin, with the remains of a scorched ashwalker village nestled between the crags. Comes with an associated area for this ruin.

A blood drunk miner defending a vault of loot!
Screenshot 2024-01-29 165828

## Why It's Good For The Game

Lava planets need ruins.

## Changelog

:cl:
add: New lava canyon ruin
fix: Gate icon updates not working
/:cl:

### CODE REVIEW

**Review by Zevotech — DISMISSED** [review]
Alright, a lavaland ruin! Finally!
Starting off strong, you did an excellent job blending the area of the ruin with the natural environment. The fact that I can't see where the planetgen ends and the ruin begins is a huge win for this ruin. This is a positive comment, no need to change anything about the ruin in terms of environment as far as I can tell.

This may surprise you, but the BDM is actually passive and minds their business until you attack them! People can just walk up and yoink that chest with no risk. I'd either remove the chest, or put it behind a locked gate with a setup similar to the ashdrake ruin (Boss drops a key to the gate) to prevent people from just snatching it. Remember that the boss itself is a pretty big reward for anyone competent enough to beat it.
image
image

Two things. One, the ashwalker corpses will get husked by the first storm that comes in which may mess with the vibes you were going for. Second, mixing these two types of walls is weird. I'd either stick with just using sandstone or necropolis walls to really send the ashie vibes home.
image

Interesting setup here. I'm not sure if this is a reference to something so I'm going to suggest changing the dusty shard to something more rewarding or removing it altogether and replacing it with a shield or something.
image

This is silly and great. I had no idea these tiles dropped out and rose back up like a platformer game. No change requested here.
image

The scrap metal/titanium feels out of place in this ruin, I'd just replace it with ore. Also having bone weapons everywhere on the floor feels weird. Maybe have them in a dedicated armory hut or just on/next to corpses.
image

Anyways this is a pretty great ruin. Hope to see it merged quickly because its in a pretty good state as-is.

File: code/game/objects/items/puzzle_pieces.dm [hunk 1]
```diff
@@ -65,6 +65,10 @@
 	desc = "A key belonging to a once peaceful scholar, brought to death and ruin through means of violence by savage outsider."
 	puzzle_id = "priestkey"
 
+/obj/item/keycard/gatedrop/lavacanyon
```
Comment (Zevotech) [inline code/game/objects/items/puzzle_pieces.dm:68]: Add the key to the blood drunk's crusher loot pool as well. Megafauna drop different things when killed with a crusher.

**Comment by Latentish** [comment]
Tming this gonna test it live =)

**Comment by Latentish** [comment]
I want to test this in game one more time. But otherwise I think this is rad as fuck.

**Review by Latentish — DISMISSED** [review]
Absolutely beautiful map. Swap out the narsien armor in the chest for something else and we can merge this.

**Review by Latentish — APPROVED** [review]
beautiful map

**Pull request merged**